Fix: add null check for activeIndicator in Carousel

PGrayCS opened the pull request:
### Description

Adds null check for activeIndicator to match the pattern used for newActiveIndicator on line 282.

### Motivation & Context

While testing carousel behavior with modified DOM, I noticed SelectorEngine.findOne() can return null if there's no active indicator, which would cause an error on classList.remove(). This adds the same safety check that's already in place for newActiveIndicator.

### Type of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (non-breaking change)
- [ ] Breaking change (fix or feature that would change existing functionality)

### Checklist

- [x] I have read the contributing guidelines
- [x] My code follows the code style of the project _(using `npm run lint`)_
- [ ] My change introduces changes to the documentation
- [ ] I have updated the documentation accordingly
- [ ] I have added tests to cover my changes
- [x] All new and existing tests passed

### Related issues

N/A

PGrayCS committed:
Fix: add null check for activeIndicator in Carousel

Prevents potential null reference error in _setActiveIndicatorElement when
no active indicator exists in the carousel.

julien-deramond commented:
> Fair question. I ran into this when the carousel was initialized programmatically and the indicators element existed but had no active indicator yet. The inconsistency is that newActiveIndicator already has this check (line 282) but activeIndicator doesn't.

Could you please provide a reproducible use case so we can test it manually, and if possible, include the corresponding unit test in this PR?

PGrayCS commented:
## Reproducible Use Case

I've created a standalone HTML test case that demonstrates the issue. The scenario occurs when:

1. A carousel is initialized with indicators present in the DOM
2. None of the indicators have the `active` class or `aria-current` attribute
3. The carousel has an active slide item
4. When sliding to another item, `_setActiveIndicatorElement()` is called
5. `SelectorEngine.findOne(SELECTOR_ACTIVE, this._indicatorsElement)` returns `null` because no indicator is marked as active
6. Without the fix, attempting to call `.classList.remove()` on `null` throws a TypeError

You can test this by:
- Creating a carousel programmatically where you manipulate the indicators after initialization
- Or in cases where the indicators are dynamically added without proper active state synchronization

I've included a standalone HTML file in this comment that demonstrates the issue. Save it and open it in a browser to test.

<details>
<summary>Click to view the test HTML file</summary>

\`\`\`html
<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <title>Carousel Indicators Test Case</title>
  <link href="https://cdn.jsdelivr.net/npm/bootstrap@5.3.0/dist/css/bootstrap.min.css" rel="stylesheet">
  <style>
    .test-section { margin: 30px 0; padding: 20px; border: 1px solid #ddd; border-radius: 8px; }
    .carousel { max-width: 600px; }
    .carousel-item { height: 300px; background-color: #f8f9fa; }
  </style>
</head>
<body>
  <div class="container py-5">
    <h1>Carousel Indicators Test Case</h1>
    
    <div class="test-section">
      <h2>Test Case: No Active Indicator Initially</h2>
      
      <div id="testCarousel" class="carousel slide">
        <div class="carousel-indicators">
          <button type="button" data-bs-target="#testCarousel" data-bs-slide-to="0" aria-label="Slide 1"></button>
          <button type="button" data-bs-target="#testCarousel" data-bs-slide-to="1" aria-label="Slide 2"></button>
        </div>
        
        <div class="carousel-inner">
          <div class="carousel-item active">
            <div class="d-flex align-items-center justify-content-center h-100 bg-primary text-white">
              <h3>Slide 1</h3>
            </div>
          </div>
          <div class="carousel-item">
            <div class="d-flex align-items-center justify-content-center h-100 bg-success text-white">
              <h3>Slide 2</h3>
            </div>
          </div>
        </div>
      </div>
      
      <button id="nextBtn" class="btn btn-primary mt-3">Go to Next Slide</button>
      <div id="status" class="mt-3"></div>
    </div>
  </div>

  <script src="https://cdn.jsdelivr.net/npm/bootstrap@5.3.0/dist/js/bootstrap.bundle.min.js"></script>
  <script>
    // Remove active class from all indicators to simulate the issue
    const indicators = document.querySelectorAll('.carousel-indicators button');
    indicators.forEach(indicator => {
      indicator.classList.remove('active');
      indicator.removeAttribute('aria-current');
    });

    document.getElementById('nextBtn').addEventListener('click', () => {
      try {
        const carousel = bootstrap.Carousel.getInstance(document.getElementById('testCarousel')) || 
                        new bootstrap.Carousel(document.getElementById('testCarousel'));
        carousel.next();
        document.getElementById('status').textContent = 'Success!';
        document.getElementById('status').style.color = 'green';
      } catch (error) {
        document.getElementById('status').textContent = 'Error: ' + error.message;
        document.getElementById('status').style.color = 'red';
      }
    });
  </script>
</body>
</html>
\`\`\`

</details>

## Unit Test

I've also added a unit test to the PR to cover this scenario:

\`\`\`javascript
it('should handle indicators without an active indicator on initialization', () => {
  return new Promise(resolve => {
    fixtureEl.innerHTML = [
      '<div id="myCarousel" class="carousel slide">',
      '  <div class="carousel-indicators">',
      '    <button type="button" data-bs-target="myCarousel" data-bs-slide-to="0" aria-label="Slide 1"></button>',
      '    <button type="button" id="secondIndicator" data-bs-target="myCarousel" data-bs-slide-to="1" aria-label="Slide 2"></button>',
      '  </div>',
      '  <div class="carousel-inner">',
      '    <div class="carousel-item active">item 1</div>',
      '    <div class="carousel-item">item 2</div>',
      '  </div>',
      '</div>'
    ].join('')

    const carouselEl = fixtureEl.querySelector('#myCarousel')
    const secondIndicator = fixtureEl.querySelector('#secondIndicator')
    const carousel = new Carousel(carouselEl)

    carouselEl.addEventListener('slid.bs.carousel', () => {
      expect(secondIndicator).toHaveClass('active')
      expect(secondIndicator.getAttribute('aria-current')).toEqual('true')
      resolve()
    })

    carousel.next()
  })
})
\`\`\`

This test verifies that when no indicator is initially active, the carousel can still slide correctly and properly activate the new indicator without throwing an error.

PGrayCS committed:
Add test for carousel indicators without active state

Covers scenario where carousel has indicators but none are marked as
active initially. Ensures sliding works correctly and activates the
new indicator without errors.

PGrayCS commented:
Sure thing. This happens when the carousel is initialized programmatically and the indicators element exists but has no active indicator yet. In that case, `SelectorEngine.findOne(SELECTOR_ACTIVE, this._indicatorsElement)` returns null, and then the code tries to call `.classList.remove()` on null which throws a TypeError.

I ran into this when I was building indicators dynamically and hadn't synced up the active states properly before the first slide transition.

I've added a test case in the latest commit that covers this scenario - creates a carousel with indicators but none marked as active, then calls `next()`. Verifies it slides correctly and activates the new indicator without erroring.

The null check matches what's already there for `newActiveIndicator` on line 282, so it keeps the code consistent.
